src/engine/GameObjects/MapNode.cxx

```diff
@@ -226,9 +226,10 @@ void MapNode::updateTexture()
           m_sprite->setClipRect({clipRect.x + m_clippingWidth * m_mapNodeData[currentLayer].tileData->tiles.offset, 0,
                                  m_clippingWidth, m_mapNodeData[currentLayer].tileData->tiles.clippingHeight},
                                 static_cast<Layer>(currentLayer));
-          bool shouldRenderTexture = m_mapNodeData[currentLayer].tileID != DEMY_NODE_ID;
-          m_sprite->setTexture(TileManager::instance().getTexture(m_mapNodeData[currentLayer].tileID),
```

Comment:
it's not possible because for that to work we need to add a check inside "updateNeighborsOfNode" because this function calls "setBitmask" which calls "updateTexture".
so in this case this function will call updateTexture for Tiles set to should not render. and cause rendering problems.
that's happen inside "updateNeighborsOfNode" which called from "setTileIDOfNode".
updateNeighborsOfNode(*begin); (begin here is the original corner Tile)
so when getting the neighbors we don't know what layer is being modified (that would cause future problems in case we edited it to check for last modified Layer in my opinion.)
updateNeighborsOfNode is called from within multiple places, decreaseHeight, increaseHeight, ...
I think you understand what I meant with all of this explanation. :P

Reply:
thanx for the thorough explanantion!